UnityProject/Assets/Scripts/NPC/AI/CorgiAI.cs

```diff
@@ -0,0 +1,190 @@
+
+		if (speaker.Script == null) return;
+		if (speaker.Script.playerNetworkActions == null) return;
+
+			// Check for an ID card (could not find a better solution)
+		var PNA = speaker.Script.playerNetworkActions;
+		IDCard card = null;
+		if (PNA.Inventory.ContainsKey(EquipSlot.id) &&
+		    PNA.Inventory[EquipSlot.id].Item?.GetComponent<IDCard>() != null)
+		{
+			card = PNA.Inventory[EquipSlot.id].Item.GetComponent<IDCard>();
+		}
+		else if (PNA.Inventory.ContainsKey(PNA.activeHand) &&
+		         PNA.Inventory[PNA.activeHand].Item?.GetComponent<IDCard>() != null)
+		{
+			card = PNA.Inventory[PNA.activeHand].Item.GetComponent<IDCard>();
+		}
+
+		if (card == null) return;
+
+		bool allowCommands = false;
+		foreach (JobType t in allowedToGiveCommands)
+		{
+			if (t == card.GetJobType) allowCommands = true;
+		}
+
+		if (!allowCommands) return;
+
+		StartCoroutine(PerformVoiceCommand(chatEvent.message.ToLower(), speaker));
+	}
+
+	IEnumerator PerformVoiceCommand(string msg, ConnectedPlayer speaker)
+	{
+		//We want these ones to happen right away:
+		if (msg.Contains($"{dogName} run") || msg.Contains($"{dogName} get out of here"))
+		{
+			StartFleeing(speaker.GameObject.transform, 10f);
+			yield break;
+		}
+
+		if (msg.Contains($"{dogName} stay") || msg.Contains($"{dogName} sit")
+		                                    || msg.Contains($"{dogName} stop"))
+		{
+			ResetBehaviours();
+			yield break;
+		}
+
+		//Slight delay for the others:
+		yield return WaitFor.Seconds(0.5f);
+
+		if (msg.Contains($"{dogName} come") || msg.Contains($"{dogName} follow"))
+		{
+			if (Random.value > 0.8f)
+			{
+				yield return ChaseTail(1);
+			}
+
+			FollowTarget(speaker.GameObject.transform);
+			yield break;
+		}
+
+		if (msg.Contains($"{dogName} find food") || msg.Contains($"{dogName} explore"))
+		{
+			if (Random.value > 0.8f)
+			{
+				yield return ChaseTail(2);
+			}
+
+			BeginExploring();
+			yield break;
+		}
+	}
+
+	IEnumerator ChaseTail(int times)
+	{
+		var timesSpun = 0;
+
+		while (timesSpun <= times)
+		{
+			for (int spriteDir = 1; spriteDir < 5; spriteDir++)
+			{
+				dirSprites.DoManualChange(spriteDir);
+				yield return WaitFor.Seconds(0.4f);
+			}
+
+			timesSpun++;
+		}
+
+		yield return WaitFor.EndOfFrame;
+	}
+
+	//TODO: Do extra stuff on these events, like barking when being told to sit:
+	void OnFleeStopped()
+	{
+	}
+
+	void OnExploreStopped()
+	{
+	}
+
+	void OnFollowingStopped()
+	{
+	}
+
+	//Updates only on the server
+	protected override void UpdateMe()
+	{
+		if (health.IsDead || health.IsCrit || health.IsCardiacArrest) return;
+
+		base.UpdateMe();
+		MonitorExtras();
+	}
+
+	void MonitorExtras()
+	{
+		//TODO monitor hunger when it is added
+
+		if (!IsPerformingTask) return;
+
+		timeWaiting += Time.deltaTime;
+		if (timeWaiting > timeForNextRandomAction)
+		{
+			timeWaiting = 0f;
+			timeForNextRandomAction = Random.Range(10f, 100f);
+
+			DoRandomAction(Random.Range(1, 3));
```

Comment:
remember that it will never pick 3
```
public static int Range(int min, int max);
Return a random integer number between min [inclusive] and max [exclusive] (Read Only).
Note max is exclusive. Random.Range(0, 10) can return a value between 0 and 9.
```